R/zchunk_L124.nonco2_unmgd_R_S_T_Y.R

```diff
@@ -43,80 +41,182 @@ module_emissions_L124.nonco2_unmgd_R_S_T_Y_DISABLED <- function(command, ...) {
+      mutate(PctForestFire = if_else( is.na(PctForestFire), 1, PctForestFire)) %>%                     # Assume missing values mean 100% forest fires since these are easier to model in GCAM
+      mutate(PctDeforest = if_else( is.na(PctDeforest), 0, PctDeforest)) %>%
+      mutate(ForestFire = value * PctForestFire) %>%                                                   # Compute forest fire emissions
+      mutate(Deforest = value * PctDeforest) %>%                                                       # Compute deforestation emissions
+      select(-value, -PctForestFire, -PctDeforest) %>%
+      gather(technology, value, -GCAM_region_ID, -GLU, -Land_Type, -Non.CO2, -year) ->
+      L124.nonco2_tg_R_forest_Y_GLU
+
+    # Compute global average deforestation emissions coefficients
+    # Compute total change in forest area from 2000 to 2005, total global emissions, and average annualized coefficients (emissions / change in land area / number of years)
+    gas_list <- tibble(Non.CO2 = L124.nonco2_tg_R_forest_Y_GLU$Non.CO2 %>% unique())                  # Set up list of gases (we'll use this to add columns later)
+    L124.LC_bm2_R_UnMgdFor_Yh_GLU_adj %>%
+      filter(year %in% emissions.DEFOREST_COEF_YEARS) %>%                                             # Get years that we'll use for deforestation calculation (as of 5/14/17 this was 2000 & 2005)
+      mutate(year = if_else(year == min(emissions.DEFOREST_COEF_YEARS), "year1", "year2")) %>%        # Rename years so we can use them as column headings (this also makes this robust to changes in years later)
+      spread(year, value) %>%                                                                         # Spread so years are separate columns
+      mutate(driver = (year1 - year2) / (emissions.DEFOREST_COEF_YEARS[2] - emissions.DEFOREST_COEF_YEARS[1])) %>%    # Compute average annual deforestation rates (change in forest area / number of years)
+      mutate(driver = if_else(driver < 0, 0, driver)) %>%                                             # Deforestation emissions only happen if forest area decreases
+      repeat_add_columns(gas_list) %>%                                                                # Add in rows for all required emissions
+      left_join(filter(L124.nonco2_tg_R_forest_Y_GLU,                                                 # Map in EDGAR deforestation emissions for the final deforestation year (as of 5/14/17 this was 2005)
+                       year == emissions.DEFOREST_COEF_YEARS[2],
+                       technology == "Deforest"), by = c( "GCAM_region_ID", "Land_Type", "GLU", "Non.CO2")) %>%
+      mutate(technology = if_else(is.na(technology), "Deforest", technology)) %>%                     # Make sure the technology name is "Deforest" (not sure why I have to do this but it is required)
+      mutate(value = if_else(is.na(value), 0, value)) %>%
```

Comment:
We've been doing this via `if_else` but `tidyr::replace_na` is about 10x as fast.